GoRogue/MapGeneration/Region.cs

```diff
@@ -9,17 +10,56 @@
 
 namespace GoRogue.MapGeneration
 {
+    /// <summary>
+    /// Event fired when a region's Area is changed.
+    /// </summary>
+    [PublicAPI]
+    public class RegionAreaChanged : EventArgs
+    {
+        /// <summary>
+        /// The region whose area is changing
+        /// </summary>
+        public readonly Region Item;
+
+        /// <summary>
+        /// The previous Area
+        /// </summary>
+        public readonly Area Old;
+
+        /// <summary>
+        /// The Area to which we update
+        /// </summary>
+        public readonly Area New;
+
+        /// <summary>
+        /// The Event Arguments for when a region's area is changed
+        /// </summary>
+        /// <param name="item">The region whose area is changing</param>
+        /// <param name="oldArea">The former value of the Area</param>
+        /// <param name="newArea">The new value of the Area</param>
+        public RegionAreaChanged(Region item, Area oldArea, Area newArea)
```

Comment:
Ideally we can name the parameters the same as their corresponding fields, just lower case starting later; eg. `oldValue` and `newValue`, if suggestions from above are implemented.  Largely this just keeps things consistent with the rest of the library.